test/suite/test_gc01.py

```diff
@@ -35,12 +35,9 @@ def timestamp_str(t):
     return '%x' % t
 
 # test_gc01.py
-# Test that checkpoint cleans the obsolete lookaside pages.
-class test_gc01(wttest.WiredTigerTestCase):
-    # Force a small cache.
-    conn_config = 'cache_size=50MB,log=(enabled)'
-    session_config = 'isolation=snapshot'
 
+# Shared base class used by gc tests.
+class test_gc_base(wttest.WiredTigerTestCase):
     def large_updates(self, uri, value, ds, nrows, commit_ts):
```

Comment:
I like that this test code is getting refactored.  Does any of the refactoring have to do with this ticket?  If not, it might be nice to put it in a separate ticket/commit.

Reply:
On a related note, I wonder if any of these test changes expose a problem with eviction that is motivating this ticket?

Reply:
I got this problem when I was testing the `test_gc03.py` test. As the testing ticket has a dependency on other tickets to let the `search` work properly. I thought of pushing the test code reconsiderations here instead of waiting. I am fine to move them to test ticket.